### ISSUE TITLE
Add a math test for reference DataVectors

### ISSUE THREAD

**Issue by fmahebert** [sxs-collaboration/spectre#83]
We have a few tests that check that `DataVectors` and reference `DataVectors` behave the same way, for instance,
```
Unit.Serialization.DataVector
Unit.Serialization.DataVector_Ref
```

We should add a test `Unit.DataStructures.DataVector.Math_Ref` to ensure that reference DataVectors behave correctly as arithmetic types.

Historical aside: in the previous repository, this check was (partially) handled within the Determinant test, but it really belongs as its own unit test.

**Comment by moxcodes** [comment]
Am I correct in understanding that this is solved as of [PR763](https://github.com/sxs-collaboration/spectre/pull/763)?

**Comment by moxcodes** [comment]
Answer from @fmahebert : no, but I (moxcodes) will fix it in datavectorimpl

**Comment by kidder** [comment]
@moxcodes has this been solved by one of your PRs?  If so, please link to the PR and close this.

**Comment by moxcodes** [comment]
Yep! This has been fixed by #1236 .
